packages/components/src/snackbar/types.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import type { MutableRefObject, ReactNode, SyntheticEvent } from 'react';
+
+export type NoticeActionWithURL = {
+	label: string;
+	url: string;
+	onClick?: ( event: SyntheticEvent ) => void;
+};
+
+type NoticeActionWithOnClick = {
+	label: string;
+	url?: string;
+	onClick: ( event: SyntheticEvent ) => void;
+};
+
+export type NoticeAction = NoticeActionWithURL | NoticeActionWithOnClick;
+
+export type Notice = {
+	id: string;
+	spokenMessage: string;
+	actions: NoticeAction[];
+	icon?: ReactNode;
+	onDismiss?: () => void;
+	content: string;
+	isDismissible: boolean;
+	explicitDismiss: boolean;
+};
+
+export type SnackbarProps = {
+	/**
+	 * The displayed message of a notice. Also used as the spoken message for assistive technology, unless `spokenMessage` is provided as an alternative message.
```

Comment:
Let's break the line and make sure that comments in this file don't go beyond the 80 characters mark in each line (`children`, `actions` and `politeness`)

Reply:
Sounds good, committed in https://github.com/WordPress/gutenberg/pull/45472/commits/70bfe595c8de58542f45de8ac4bb7b90fd31e75a